test/test_evaluator.cpp

```diff
@@ -696,12 +696,13 @@ TEST_F(ExprEvalTest, MmaOp) {
 
   at::Tensor in_a = at::ones(a_shape, at::kHalf).cuda();
   at::Tensor in_b = at::ones(b_shape, at::kHalf).cuda();
-  at::Tensor out_ref = k * at::ones(out_shape, at::kHalf).cuda();
+  auto out_dtype = data_type_to_aten(tv2->getDataType().value());
+  at::Tensor out_ref = k * at::ones(out_shape, out_dtype).cuda();
```

Comment:
Why did you make this change? The old code looks better. In general, make the reference implementation as straightforward as possible, and avoid reusing stuff from the actual implementation. This is because the part you reuse could be buggy.